qiskit/providers/aer/backends/cluster/clusterresults.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Results managed by the Job Manager."""
+
+from typing import List, Optional, Union, Tuple, Dict
+import copy
+
+# TODO Use TYPE_CHECKING instead of pylint disable after dropping python 3.5
+import numpy  # pylint: disable=unused-import
+from qiskit.result import Result
+from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import Schedule
+
+from qiskit.providers import JobError
+from .exceptions import AerClusterResultDataNotAvailable
+
+
+class CResults:
```

Comment:
We shouldn't need a custom result class.

Handling should be done via the `AerJob` and `AerJobSet` classes, so that each worker is described by an `AerJob` which just returns a regular `Result` object. The AerJobSet class should have functionality for combining all the individual worker job results into a single `Result` object.